File: packages/go/qorechain/networks/networks.go
// Package networks provides built-in network presets for the QoreChain Go SDK.
//
// Both the "testnet" and "mainnet" presets are fully populated and live; their
// endpoints default to localhost ports so the SDK works out of the box against a
// locally running node, and callers can override them with real hostnames.
// GetNetwork returns either preset.
package networks

import "fmt"

// Bech32Prefixes holds the bech32 human-readable prefixes used across
// QoreChain address types.
type Bech32Prefixes struct {
	Account   string
	Validator string
	Consensus string
}

// CoinInfo holds display and base denomination metadata for the network's
// staking coin.
type CoinInfo struct {
	Display  string
	Base     string
	Exponent int
}

// Endpoints holds the service endpoints for talking to a network across its
// supported VMs.
type Endpoints struct {
	REST   string
	GRPC   string
	RPC    string
	EVMRPC string
	EVMWS  string
	SVMRPC string
}

// NetworkConfig is a fully described network preset.
type NetworkConfig struct {
	Name      string
	Live      bool
	ChainID   string
	Bech32    Bech32Prefixes
	Coin      CoinInfo
	Endpoints *Endpoints
}

// QoreChain uses the same token and address prefixes on every network.
var (
	bech32Prefixes = Bech32Prefixes{Account: "qor", Validator: "qorvaloper", Consensus: "qorvalcons"}
	coinInfo       = CoinInfo{Display: "QOR", Base: "uqor", Exponent: 6}
)

// Networks is the set of built-in network presets, keyed by name.
var Networks = map[string]NetworkConfig{
	"testnet": {
		Name:    "testnet",
		Live:    true,
		ChainID: "qorechain-diana",
		Bech32:  bech32Prefixes,
		Coin:    coinInfo,
		Endpoints: &Endpoints{
			REST:   "http://localhost:1317",
			GRPC:   "http://localhost:9090",
			RPC:    "http://localhost:26657",
			EVMRPC: "http://localhost:8545",
			EVMWS:  "ws://localhost:8546",
			SVMRPC: "http://localhost:8899",
		},
	},
	"mainnet": {
		Name:    "mainnet",
		Live:    true,
		ChainID: "qorechain-vladi",
		Bech32:  bech32Prefixes,
		Coin:    coinInfo,
		Endpoints: &Endpoints{
			REST:   "http://localhost:1317",
			GRPC:   "http://localhost:9090",
			RPC:    "http://localhost:26657",
			EVMRPC: "http://localhost:8545",
			EVMWS:  "ws://localhost:8546",
			SVMRPC: "http://localhost:8899",
		},
	},
}

// GetNetwork resolves a network preset by name.
//
// It returns an error if the named network is unknown.
func GetNetwork(name string) (NetworkConfig, error) {
	config, ok := Networks[name]
	if !ok {
		return NetworkConfig{}, fmt.Errorf("unknown network: %s", name)
	}
	return config, nil
}

// ListNetworks lists the known network preset names without any liveness check.
func ListNetworks() []string {
	return []string{"testnet", "mainnet"}
}

File: packages/py/src/qorechain/networks.py
"""Network presets for the QoreChain Python SDK.

The ``testnet`` preset is fully populated and live; its endpoints default to
localhost ports so the SDK works out of the box against a locally running node,
and callers can override them with real hostnames. The ``mainnet`` preset is a
placeholder: mainnet is not yet live, so it carries no chain ID and no
endpoints, and :func:`get_network` raises if asked for it.
"""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class Bech32Prefixes:
    """Bech32 human-readable prefixes used across QoreChain address types."""

    account: str
    validator: str
    consensus: str


@dataclass(frozen=True)
class CoinInfo:
    """Display and base denomination metadata for the network's staking coin."""

    display: str
    base: str
    exponent: int


@dataclass(frozen=True)
class NetworkEndpoints:
    """Service endpoints for talking to a network across its supported VMs."""

    rest: str
    grpc: str
    rpc: str
    evm_rpc: str
    evm_ws: str
    svm_rpc: str


@dataclass(frozen=True)
class NetworkConfig:
    """A fully described network preset."""

    name: str
    live: bool
    chain_id: str | None
    bech32: Bech32Prefixes
    coin: CoinInfo
    endpoints: NetworkEndpoints | None


# QoreChain uses the same token and address prefixes on every network.
_BECH32 = Bech32Prefixes(account="qor", validator="qorvaloper", consensus="qorvalcons")
_COIN = CoinInfo(display="QOR", base="uqor", exponent=6)

#: The set of built-in network presets, keyed by name.
NETWORKS: dict[str, NetworkConfig] = {
    "testnet": NetworkConfig(
        name="testnet",
        live=True,
        chain_id="qorechain-diana",
        bech32=_BECH32,
        coin=_COIN,
        endpoints=NetworkEndpoints(
            rest="http://localhost:1317",
            grpc="http://localhost:9090",
            rpc="http://localhost:26657",
            evm_rpc="http://localhost:8545",
            evm_ws="ws://localhost:8546",
            svm_rpc="http://localhost:8899",
        ),
    ),
    "mainnet": NetworkConfig(
        name="mainnet",
        live=False,
        chain_id=None,
        bech32=_BECH32,
        coin=_COIN,
        endpoints=None,
    ),
}


def get_network(name: str) -> NetworkConfig:
    """Resolve a network preset by name.

    :returns: The requested live :class:`NetworkConfig`.
    :raises ValueError: If the named network is unknown or not yet live
        (e.g. ``mainnet``). Pass custom endpoints to target such a network.
    """
    config = NETWORKS.get(name)
    if config is None:
        raise ValueError(f"unknown network: {name}")
    if not config.live:
        raise ValueError(f"{name} is not yet live — pass custom endpoints")
    return config


def list_networks() -> list[str]:
    """List the known network preset names without any liveness check."""
    return list(NETWORKS.keys())
